pkgs/os-specific/linux/dcgm/default.nix

```diff
@@ -0,0 +1,108 @@
+{ lib
+, callPackage
+, gcc11Stdenv
+, fetchFromGitHub
+, addOpenGLRunpath
+, catch2
+, cmake
+, cudaPackages_10_2
+, cudaPackages_11_8
+, cudaPackages_12
+, fmt_9
+, git
+, jsoncpp
+, libevent
+, plog
+, python3
+, tclap_1_4
+, yaml-cpp
+}:
+let
+  # Flags copied from DCGM's libevent build script
+  libevent-nossl = libevent.override { sslSupport = false; };
+  libevent-nossl-static = libevent-nossl.overrideAttrs (super: {
+    CFLAGS = "-Wno-cast-function-type -Wno-implicit-fallthrough -fPIC";
+    CXXFLAGS = "-Wno-cast-function-type -Wno-implicit-fallthrough -fPIC";
+    configureFlags = super.configureFlags ++ [ "--disable-shared" "--with-pic" ];
+  });
+
+  jsoncpp-static = jsoncpp.override { enableStatic = true; };
+
+  cudaToolkits = [
+    {
+      version = "10";
+      toolkit = cudaPackages_10_2.cudatoolkit;
+    }
+    {
+      version = "11";
+      toolkit = cudaPackages_11_8.cudatoolkit;
+    }
+    {
+      version = "12";
+      toolkit = cudaPackages_12.cudatoolkit;
+    }
+  ];
+
+  # Builds CMake code to add CUDA paths for include and lib.
+  mkAppendCudaPaths = { version, toolkit }: ''
+    list(APPEND Cuda${version}_INCLUDE_PATHS "${toolkit}/include")
+    list(APPEND Cuda${version}_LIB_PATHS "${toolkit}/lib" "${toolkit.lib}/lib")
+  '';
+in gcc11Stdenv.mkDerivation rec {
+  pname = "dcgm";
+  version = "3.1.8";
+
+  src = fetchFromGitHub {
+    owner = "NVIDIA";
+    repo = "DCGM";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-OXqXkP2ZUNPzafGIgJ0MKa39xB84keVFFYl+JsHgnks=";
+  };
+
+  # Add our paths to the CUDA paths so FindCuda.cmake can find them.
+  EXTRA_CUDA_PATHS = lib.strings.concatMapStringsSep "\n" mkAppendCudaPaths cudaToolkits;
+  prePatch = ''
+    echo "$EXTRA_CUDA_PATHS"$'\n'"$(cat cmake/FindCuda.cmake)" > cmake/FindCuda.cmake
+  '';
+
+  hardeningDisable = [ "all" ];
```

Comment:
why is this necessary?

Reply:
I tried disabling only some hardenings with no success. I did not see a pattern in Nixpkgs of documenting the exact failures that exist with each hardening mode. That would be a significant amount of work and it would likely bitrot very quickly.

Reply:
ok, fair enough